relayer/relayer.go

```diff
@@ -99,22 +99,11 @@ func NewRelayer(
 		return nil, err
 	}
 
-	var filteredDestinationClients map[ids.ID]vms.DestinationClient
-	supportedDestinationsBlockchainIDs := sourceSubnetInfo.GetSupportedDestinations()
-	if len(supportedDestinationsBlockchainIDs) > 0 {
-		filteredDestinationClients := make(map[ids.ID]vms.DestinationClient)
-		for id := range supportedDestinationsBlockchainIDs {
-			filteredDestinationClients[id] = destinationClients[id]
-		}
-	} else {
-		filteredDestinationClients = destinationClients
-	}
```

Comment:
I'm not entirely sure why, but this was resulting in an empty `filteredDestinationClients` with the rest of the changes in this PR. I didn't look too deeply into it because this filtering is done in config.go anyway.

Reply:
I think it's because we were creating a new local `filteredDestinationClients` variable in the `if` clause by using ":=" instead of "="

Reply:
Gah that's definitely it. I'm surprised the linter didn't pick up `filteredDestinationClients` as unused, but maybe that's because it's being passed mutably to a function.